monster-cards/src/stores/cardStore.ts

```diff
@@ -78,9 +78,49 @@ export interface ChangeRecord {
 export function bindActions() {
 	return cardStore.observe().subscribe((options: any) => {
 			const changeRecord = <ChangeRecord> options;
-
 			return putCard.do(changeRecord);
 	});
 }
 
+function shuffle (array: any[]): any[] {
+	let i = 0;
+	let	j = 0;
+	let	temp: any;
+
+	for (i = array.length - 1; i > 0; i -= 1) {
+		j = Math.floor(Math.random() * (i + 1));
+		temp = array[i];
+		array[i] = array[j];
+		array[j] = temp;
+	}
+
+	return array;
+}
+
+export function pickRandomCards(numberToPick: number, exclude: string[] = []): Promise<Card[]> {
```

Comment:
this seems to be a little over complicated? we go from an iterable to an array, to a set, and then back to an array. And we have a for loop with a do while inside.

```javascript
return cardStore.get().then((cards) => {
    const filteredCards = Array.from(cards).filter((card) => !exclude.includes(card));
    const shuffledCards = shuffle(filteredCards);
    return shuffledCards.slice(0, Math.min(shuffledCards.length, numberToPick));
});
```